test/infrastructure/docker/docker/machine.go

```diff
@@ -317,6 +317,33 @@ func (m *Machine) ExecBootstrap(ctx context.Context, data string) error {
 	return nil
 }
 
+// CheckForBootstrapSuccess checks if bootstrap was successful by checking for existence of the sentinel file.
+func (m *Machine) CheckForBootstrapSuccess(ctx context.Context) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	if m.container == nil {
+		return errors.New("unable to set CheckForBootstrapSuccess. the container hosting this machine does not exists")
+	}
+
+	command := cloudinit.Cmd{
+		Cmd:  "test",
+		Args: []string{"-f", "/run/cluster-api/bootstrap-success.complete"},
+	}
```

Comment:
nit. you could avoid creating a `cloudinit.Cmd` and pass cmd and args directly to `m.container.Commander.Command`